### PR TITLE
Support some ONNX operators

### PR DESCRIPTION
**Pull request by tagomaru**
Signed-off-by: tagomaru <tagomaru@users.noreply.github.com>

This PR is going to support some ONNX operators.

### CODE REVIEW

File: maraboupy/MarabouNetworkONNX.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 from maraboupy import MarabouNetwork
 from onnx import TensorProto
 import itertools
+import torch
```
Comment (anwu1219) [inline maraboupy/MarabouNetworkONNX.py:26]: Why do we need torch here?

File: maraboupy/MarabouNetworkONNX.py [hunk 2]
```diff
@@ -761,13 +768,165 @@ def matMulEquations(self, node, makeEquations):
...
+        if not makeEquations:
+            return
+
+        # Get variables
+        inputVars = np.concatenate(inputVars, axis).reshape(-1)
+        outputVars = self.makeNewVariables(nodeName).reshape(-1)
+
+        assert len(inputVars) == len(outputVars)
+
+        for i in range(len(inputVars)):
```
Comment (anwu1219) [inline maraboupy/MarabouNetworkONNX.py:801]: For these operations, I don't think we need to actually introduce new equations or variables, which makes the verification query harder. Please the reshape, flatten, or transpose methods in this class.

File: maraboupy/MarabouNetworkONNX.py (hunk 1 above)
Comment (tagomaru) [reply maraboupy/MarabouNetworkONNX.py:26]: Yes, it is used.

File: maraboupy/MarabouNetworkONNX.py (hunk 2 above)
Comment (tagomaru) [reply maraboupy/MarabouNetworkONNX.py]: fixed.

File: maraboupy/MarabouNetworkONNX.py [hunk 3]
```diff
@@ -761,13 +768,149 @@ def matMulEquations(self, node, makeEquations):
...
+        Args:
+            node (node): ONNX node representing the Resize operation
+            makeEquations (bool): True if we need to create new variables and write Marabou equations
+
+        :meta private:
+        """
+        nodeName = node.output[0]
+        inputName = node.input[0]
+
+        # Check numper of dimension of input
```
Comment (anwu1219) [inline maraboupy/MarabouNetworkONNX.py:853]: number

File: maraboupy/MarabouNetworkONNX.py [hunk 4]
```diff
@@ -761,13 +768,149 @@ def matMulEquations(self, node, makeEquations):
...
+        inputVars = self.varMap[inputName]
+        inputShape = inputVars.shape
+        if inputVars.ndim != 4:
+            raise NotImplementedError("Marabou only supports resize operator for very specific upsample case used in YOLO now.")
```
Comment (anwu1219) [inline maraboupy/MarabouNetworkONNX.py:857]: Could you add a comment here describing the currently unsupported functionality of resize?

**Review by anwu1219 — APPROVED** [review]

File: maraboupy/MarabouNetworkONNX.py (hunk 3 above)
Comment (tagomaru) [reply maraboupy/MarabouNetworkONNX.py]: fixed

File: maraboupy/MarabouNetworkONNX.py (hunk 4 above)
Comment (tagomaru) [reply maraboupy/MarabouNetworkONNX.py:857]: added

**Pull request merged**